SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialEventTracer.h

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Interop/Connection/SpatialSpanIdCache.h"
+#include "Interop/Connection/SpatialTraceEvent.h"
+
+// Documentation for event tracing in the GDK can be found here: https://brevi.link/gdk-event-tracing-documentation
+
+// TODO(EventTracer): make sure SpatialEventTracer doesn't break the LatencyTracer functionality for now (maybe have some macro/branching in
+// .cpp file, when the LatencyTracer is enabled?)
+// TODO(EventTracer): make sure the overhead of SpatialEventTracer is minimal when it's switched off
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialEventTracer, Log, All);
+
+// TODO - Individual RPC Calls (distinguishing between GDK and USER)
+// TODO - RPCs on newly created objects go through a different flow and need to be handled
+
+namespace worker
+{
+namespace c
+{
+struct Io_Stream;
+struct Trace_Item;
+} // namespace c
+} // namespace worker
+
+// Note(EventTracer): EventTracer must be created prior to WorkerConnection, since it has to be passed to ConnectionConfig
+// (see SpatialConnectionManager diff)
+
+namespace SpatialGDK
+{
+// Note: SpatialEventTracer wraps Trace_EventTracer related functionality
+// It is owned by the connection handler and ViewCoordinator.
+// SpatialNetDriver initializes SpatialSender and SpatialReceiver with pointers to EventTracer read from SpatialWorkerConnection.
+// Note(EventTracer): SpatialEventTracer is supposed to never be null in SpatialWorkerConnection, SpatialSender, SpatialReceiver. Make sure
```

Comment:
In which case we should enforce this through the interface. If something shouldn't be null we should either pass it as a reference or assert on it being non null.
If the class isn't small it's often worth having a comment saying what kind of a pointer this is meant to be.

Reply:
It can be null in tests is the issue. Perhaps we need to make an event tracer in tests? Otherwise yes, being a reference would be nice.